lib/api/2.0/schemas2.js

```diff
@@ -26,26 +28,25 @@ var schemasIdGet = controller(function(req) {
 
 // GET /api/2.0/schemas/tasks
 var taskSchemasGet = controller(function () {
-    return taskOptionValidator.getAllSchemaNames({ includeNameSpace: false });
+    return workflowApiService.getTaskDefinitions()
+    .then(function(tasks) {
```

Comment:
@yyscamper I like to wrap promise returning function calls in `Promise.try` for this reason.  This normalizes promises to Bluebird style.  It also will handle any synchronous errors that might be thrown by a promise returning function.